source/macros.tex

```diff
@@ -724,6 +722,19 @@
  \end{bnfbase}
 }
 
+% "nc" are the non-copied versions of bnf environments; the non-"nc"
+% versions are copied to the grammar extraction file.
+\NewEnviron{bnf}{\begin{ncbnf}%
+\BODY%
+\gramWrite{\string\begin{ncbnf}\meaningbody\BODY\string\end{ncbnf}}%
+\end{ncbnf}}{}
+
+\NewEnviron{simplebnf}{\begin{ncsimplebnf}%
+\BODY%
+\gramwrite{\string\begin{ncsimplebnf}\meaningbody\BODY\string\end{ncsimplebnf}}%
```

Comment:
```suggestion
\gramWrite{\string\begin{ncsimplebnf}\meaningbody\BODY\string\end{ncsimplebnf}}%
```

Reply:
So I suppose we should rename `ncsimplebnf` and `ncrebnf`? We can drop the "nc" there if we are reasonably sure that `bnf` is the only environment we'll ever want extracted.

Reply:
I think it's plausible to assume we only ever want "bnf" to be extracted; everything else are either grammar quotes or non-C++ grammar things (such as regex).